src/computability/timed/lemmas.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2022 Tomaz Gomes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tomaz Gomes
+-/
+import data.nat.log
+import tactic.linarith
+/-!
+# lemmas
+  This file defines and proves some lemmas about the logarithm function (with base set fixed to 2)
+  and other arithmetic relations that will be used to prove merge sort's time complexity.
+## Main Results
+  - log_pred : ∀ (a : ℕ) , nat.log 2 a - 1 = nat.log 2 (a / 2)
+  - log_2_times : ∀ (a : ℕ), 2 * nat.log 2 (a + 2) ≤ a + 2
+-/
+
+lemma log_pred : ∀ (a : ℕ) , nat.log 2 a - 1 = nat.log 2 (a / 2)
+| 0 := by simp only [nat.log_zero_right, nat.zero_div]
+| 1 := by norm_num
+| (a + 2) :=
+begin
+  rw nat.log,
+  split_ifs,
+  { simp only [nat.add_succ_sub_one, add_zero], },
+  simp only [le_refl, not_true, zero_le, nat.one_lt_bit0_iff, and_self, le_add_iff_nonneg_left]
+       at h,
+  cases h,
+end
+
+lemma log_2_val : nat.log 2 2 = 1 :=
```

Comment:
And this is a consequence of a more general lemma that should go in `data.nat.log`. 

```lean
lemma log_base {b : ℕ} (hb : 1 < b) : nat.log b b = 1 :=
by simpa using nat.log_pow hb 1

lemma log_2_val : nat.log 2 2 = 1 :=
log_base (by norm_num)
```